Design/ssa-conversion.md

```diff
@@ -0,0 +1,53 @@
+# Converting an ExpEmbedding into SSA form
+
+In order to translate pure Kotlin functions into Viper functions their body must be translated into a single expression. Therefore, reassignments of variables may not happen on the Viper level which is why the ExpEmbedding must be translated into SSA form.
+
+## Converting a Block to SSA form
```

Comment:
Keep in mind that things are not necessarily quite so simple.  For example, an addition may have blocks on both sides.  (In general: assignments may appear as subexpressions.)  It would be good to think about how this should fit together.